src/main/java/fr/insalyon/citi/golo/compiler/ParseTreeToGoloIrVisitor.java

```diff
@@ -35,6 +36,10 @@
 
   private GoloCompilationException.Builder exceptionBuilder;
 
+  private static String syntheticName(String base) {
+    return "__$$_" + base + "_" + System.currentTimeMillis();
```

Comment:
I'm the one to blame, as Yannick said I've done that in other places for synthetic variables.

Perhaps we should just eliminate all these time-based generations in the whole code, using simple atomic long integers. Thoughts?